internal/log/log.go

```diff
@@ -38,13 +40,21 @@ var (
 	// lvlstrs is the lookup table of log levels to strings.
 	lvlstrs = level.Default()
 
-	// Syslog output, only set if enabled.
+	// syslog output, only set if enabled.
 	sysout *syslog.Writer
 
 	// timefmt is the logging time format used.
 	timefmt = "02/01/2006 15:04:05.000"
+
+	// ctxhooks allows modifying log content based on context.
+	ctxhooks []func(context.Context, *byteutil.Buffer)
 )
 
+// Hook adds the given hook to the global logger context hooks stack.
+func Hook(hook func(ctx context.Context, buf *byteutil.Buffer)) {
```

Comment:
The way this one works right now allows you to manually append anything to what will get logged. I'm wondering if it might be enough to only let you add in additional fields? That way you always have to provide a key-value pair so the log format stays consistent for anything parsing it.

Reply:
So something like `func Hook(hook func(ctx context.Context) []kv.Fields){}` instead that would then get appended into `kv.Fields` in `logf`.

Reply:
you are right that would make more sense :p